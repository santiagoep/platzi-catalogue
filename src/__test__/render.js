/* istanbul ignore file */
/* eslint-disable import/no-extraneous-dependencies */
import { Provider } from 'react-redux';
import { mount, shallow } from 'enzyme';
import Renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import { makeMockedStore } from '@store/index';
import mockedStore from '@__mocks__/store/index';
import lightTheme from '@assets/styles/themes/light';

const WrapperComponent = ({ children, config, initialState }) => {
  if (config && config.theme && config.redux) {
    const store = makeMockedStore(initialState || mockedStore);
    return (
      <Provider store={store}>
        <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>
      </Provider>
    );
  }
  if (config && config.redux) {
    const store = makeMockedStore(initialState || mockedStore);
    return <Provider store={store}>{children}</Provider>;
  }
  if (config && config.theme) {
    return <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>;
  }
  return children;
};

const renderWithProviders = (
  { redux, theme } = {},
  initialState,
  getMockedStore
) => {
  const renderProviders = (Component) => (
    <WrapperComponent
      config={{ redux, theme }}
      initialState={initialState}
      getMockedStore={getMockedStore}
    >
      {Component}
    </WrapperComponent>
  );

  return {
    mount: (Component) => mount(renderProviders(Component)),
    render: (Component) => render(renderProviders(Component)),
    shallow: (Component) => shallow(renderProviders(Component)),
    renderer: (Component) => Renderer.create(renderProviders(Component))
  };
};

export default renderWithProviders;
