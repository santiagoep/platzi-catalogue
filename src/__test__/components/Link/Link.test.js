import render from '../../render';
import CustomLink from '@components/Link/Link';

describe('<Link />', () => {
  it('should render snapshot', () => {
    const { renderer } = render();
    const component = renderer(<CustomLink href='/'>Home</CustomLink>);
    expect(component).toMatchSnapshot();
  });
  it('should click', () => {
    const spy = jest.fn();
    const { mount } = render();
    const component = mount(
      <CustomLink href='/' onClick={spy}>
        Home
      </CustomLink>
    );
    component.find('a').last().simulate('click');
    expect(spy).toHaveBeenCalled();
  });
  it('should click with external url', () => {
    const spy = jest.fn();
    const { mount } = render();
    const component = mount(
      <CustomLink href='https://www.google.com/' onClick={spy}>
        Google
      </CustomLink>
    );
    component.simulate('click');
    expect(spy).toHaveBeenCalled();
  });
});
