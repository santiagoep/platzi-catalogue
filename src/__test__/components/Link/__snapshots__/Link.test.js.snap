// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> should render snapshot 1`] = `
<div
  dataTestId=""
  href="/"
  passHref={false}
>
  <a
    className=""
  >
    Home
  </a>
</div>
`;
